include/types/layout.hpp

```diff
@@ -56,35 +53,79 @@ using RowMajor = MatrixLayout<kRow, kCol, kStride, 1>;
 template <const int kRow, const int kCol, const int kStride = kRow>
 using ColMajor = MatrixLayout<kRow, kCol, 1, kStride>;
 
+/// @brief Swizzled layout for 16x16 BaseTile.
+template <const int kElemBits, typename AtomLayout>
```

Comment:
Will `SwizzledRowMajor` be treated as an internal type and not be available for user usage (since this involves the cute Layout)?

Reply:
As we discussed, swizzling is strictly performed within the 16x16 index space. The `SwizzledRowMajor` class operates exclusively on a 16x16 set of indices.

To achieve a swizzled shared memory layout for the entire shared memory tile, the macro kernel must iterate over these 16x16 tiles, which prevents users from arbitrarily applying swizzling outside this context, at least in the current implementation.

Therefore, yes, in this implementations, `SwizzedRowMajor` is an internal layout type.

Reply:
swizzled shared layout is specified like this:

```cpp
static constexpr bool kUseSwizzledLayout = true;
using DstTile =
        SharedTile<Element, tl::RowMajor<32, 32>, kUseSwizzledLayout>;
```

Reply:
> swizzled shared layout is enabled through a boolean value.
> 
> ```c++
> static constexpr bool kUseSwizzledLayout = true;
> using DstTile =
>         SharedTile<Element, tl::RowMajor<32, 32>, kUseSwizzledLayout>;
> ```

Yes, I got it.